industrial_ci/src/tests/source_tests.sh

```diff
@@ -97,6 +97,96 @@ function run_clang_tidy_check {
     fi
 }
 
+function ici_combine_cpp_reports {
+  ici_install_pkgs_for_command lcov lcov
+  local target_ws=$1
+  cd "$target_ws" || return 1
+  # Capture initial coverage info
+  lcov --capture --initial \
+       --directory build \
+       --output-file initial_coverage.info | grep -ve "^Processing"
+  # Capture tested coverage info
+  lcov --capture \
+       --directory build \
+       --output-file test_coverage.info | grep -ve "^Processing"
+  # Combine two report (exit function  when none of the records are valid)
+  lcov --add-tracefile initial_coverage.info \
+       --add-tracefile test_coverage.info \
+       --output-file coverage.info || return 0 \
+    && rm initial_coverage.info test_coverage.info
+  # Extract repository files
+  lcov --extract coverage.info "$(pwd)/src/$TARGET_REPO_NAME/*" \
+       --output-file coverage.info | grep -ve "^Extracting"
+  # Filter out test files
+  lcov --remove coverage.info "*/test/*" \
+       --output-file coverage.info | grep -ve "^removing"
```

Comment:
Automatically filter out **gtest** and **gmock** files under the **test/** folder here.